packages/server/src/middleware/modern.js

```diff
@@ -11,12 +13,33 @@ const isModernBrowser = (ua) => {
   })
 }
 
-export default function (req, res, next) {
-  const { socket = {}, headers } = req
-  if (socket.modernMode === undefined) {
-    const ua = headers && headers['user-agent']
-    socket.modernMode = isModernBrowser(ua)
+let detected = false
+
+const detectModernBuild = ({ options, resources }) => {
+  if (detected === false && ![false, 'client', 'server'].includes(options.modern)) {
```

Comment:
Can't we relax condition to `typeof options.modern !== `undefined`?

Reply:
I think what you mean is `modern === undefined`?

If user specified `modern:true`, we also need to detect the mode 😄